platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/tipset/TipsetEventCreatorImplTests.java

```diff
@@ -787,4 +787,67 @@ void frozenEventCreationBug() {
         // We should be able to create an event on the new parent.
         assertNotNull(eventCreator.maybeCreateEvent());
     }
+
+    /**
+     * Event from nodes not in the address book should not be used as parents for creating new events. 
+     */
+    @Test
+    @DisplayName("Not Registering Events From NodeIds Not In AddressBook")
+    void notRegisteringEventsFromNodesNotInAddressBook() {
+        final Random random = getRandomPrintSeed();
+
+        final int networkSize = 4;
+
+        final AddressBook addressBook = new RandomAddressBookGenerator(random)
+                .setCustomWeightGenerator(x -> 1L)
+                .setSize(networkSize)
+                .build();
+
+        final FakeTime time = new FakeTime();
+
+        final NodeId nodeA = addressBook.getNodeId(0); // self
+        final NodeId nodeB = addressBook.getNodeId(1);
+        final NodeId nodeC = addressBook.getNodeId(2);
+        final NodeId nodeD = addressBook.getNodeId(3);
+        // Node 4 (E) is not in the address book.
+        final NodeId nodeE = new NodeId(4);
```

Comment:
The random address book generator is not using contiguous node IDs, so node 4 might actually be a valid ID. 
```suggestion
        final NodeId nodeE = new NodeId(nodeD.id() + 1);
```